octobot_trading/modes/abstract_trading_mode.py

```diff
@@ -137,8 +154,19 @@ async def stop(self) -> None:
             await producer.stop()
         for consumer in self.consumers:
             await consumer.stop()
+        await self.close_caches()
         self.exchange_manager = None
 
+    async def close_caches(self, reset_cache_db_ids=False):
+        for tentacle_name in [self.get_name()] + [evaluator.get_name() for evaluator in self.called_nested_evaluators]:
+            await databases.CacheManager().close_cache(
+                tentacle_name,
+                self.exchange_manager.exchange_name,
+                common_constants.UNPROVIDED_CACHE_IDENTIFIER if self.get_is_symbol_wildcard() else self.symbol,
+                common_constants.UNPROVIDED_CACHE_IDENTIFIER if self.get_is_time_frame_wildcard() else self.time_frame,
+                reset_cache_db_ids=reset_cache_db_ids
+            )
```

Comment:
It seems that this method is also reconstructing cache names, it don't think we should duplicate this code that is probably also used somewhere else
> right, I'm updating this part to use the same code as in evaluators

I don't think this is a good idea to make a dependance between the trading mode close and evaluators (they should be completely independent)
> here the original need is to close evaluator caches that might have been opened in nested evaluator calls while the evaluators themselves are not activated as "standalone evaluators" they will never be stopped (because never started) and therefore their cache won't be closed. Maybe we should but this into an octobot-evaluators util function ?

Reply:
Maybe. The problem here could also be the way we use standalone Evaluator. In my opinion there is no reason why something else than thé Evaluator itself should care about it's own cache